docs/_ruleset_functions.md

```diff
@@ -1,33 +1,270 @@
+
+Returns:
+- **opaque_surface_type**: The Opaque Surface Type [roof, heated slab-on-grade, unheated slab-on-grade, floor, above-grade wall, below-grade wall].  
+
+
+Logic:  
+
+- If surface tilt is more than or equal to 0 degree and less than 60 degrees, surface is classified as roof: ```if 0 <= surface.tilt < 60: surface_type = "ROOF"```  
+
+- Else if surface tile is more than 120 degrees and less than or equal to 180 degrees, surface is classified as floor: ```else if 120 < surface.tilt <= 180:```  
+
+  - Determine if surface is heated slab-on-grade: ```if ( surface.construction.has_radiant_heating ) AND ( surface.adjacent_to == "GROUND" ): surface_type = "HEATED SLAB-ON-GRADE"```  
+
+  - Else determine if surface is unheated slab-on-grade: ```else if surface.adjacent_to == "GROUND": surface_type = "UNHEATED SLAB-ON-GRADE"```  
+
+  - Else, surface is floor: ```else: surface_type = "FLOOR"```  
+
+- Else, surface is classified as wall: ```else:```  
+
+  - Determine if surface is below grade wall: ```if surface.adjacent_to == "GROUND": surface_type = "BELOW-GRADE WALL"```  
+
+  - Else determine if surface is above grade wall: ```else: surface_type = "ABOVE-GRADE WALL"```  
+
+**Returns** ```return surface_type```  
+
+## get_wwr
+
+Description: This function would determine window wall ratio for a building segment.  
+
+Inputs:
+  - **RMR**: The RMR that needs to determine window wall ratio.  
+
+Returns:
+- **window_wall_ratio**: The window wall ratio of the building segment.  
+
+Function Call:
+
+- get_zone_conditioning_category()
+- get_surface_conditioning_category()
+- get_opaque_surface_type()
+
+Logic:
+
+- Get zone conditioning category dictionary for RMR: `zcc_dictionary = get_zone_conditioning_category(RMR)`
+
+- Get surface conditioning category dictionary for RMR: `scc_dictionary = get_surface_conditioning_category(RMR)`
+
+- For each building segment in the model: `for building_segment in RMR.building.building_segments:`
+
+  - For each thermal_block in building segment: `for thermal_block in building_segment.thermal_blocks:`
+
+    - For each zone in thermal block: `zone in thermal_block.zones:`
+
+      - Check if zone is conditioned or semi-heated: `if zcc_dictionary[zone.id] in [CONDITIONED RESIDENTIAL, CONDITIONED NON-RESIDENTIAL, CONDITIONED MIXED, SEMI-HEATED]`
+
+        - For each surface in zone: `for surface in zone.surfaces:`
+
+          - If surface is above-grade wall and adjacent to the exterior: `if ((get_opaque_surface_type(surface) == "ABOVE-GRADE WALL") AND (surface.adjacent_to == "EXTERIOR")):`  
+
+            - For each subsurface in surface: `for subsurface in surface.subsurfaces:`  
+
+              - If the glazed vision area is less than 50% of the total area, add glazed vision area to building segment total fenestration area: `if ( subsurface.glazed_area < ( subsurface.glazed_area + subsurface.opaque_area ) * 50% ): total_fenestration_area += subsurface.glazed_area`  
+
```

Comment:
From 90.1 Section 3: door where one-half or less of the door area is glazed is considered an opaque door. 

So I think the logic should be to calculate (subsurface.glazed_area + subsurface.opaque_area) for all subsurfaces that have qualified parent surfaces (i.e., exterior or semi-exterior walls) AND (have subsurface.classification != "door" OR (subsurface.classification=door AND (subsurface.glazes_area=>0.5*(subsurface.glazed_area + subsurface.opaque_area))